core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -200,10 +201,13 @@ private[spark] class TaskSetManager(
   private[scheduler] var localityWaits = myLocalityLevels.map(getLocalityWait)
 
   // Delay scheduling variables: we keep track of our current locality level and the time we
-  // last launched a task at that level, and move up a level when localityWaits[curLevel] expires.
+  // last launched a task at that level and were also fully utilizing all slots
```

Comment:
nit: period at end of sentence.

I'd also reference here that the idea of "all slots" is dependent on the scheduling mode and more details are in `Pool.updateAvailableSlots()` (and as mentioned previously we need more comments there).